infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -579,5 +602,45 @@ public boolean isConnectionReceiver() {
     public void notifyLeadershipChange() {
         incomingMsgHandler.leadershipChanged();
     }
+
+    public void createAndSendOutgoingSession(ReplicationSubscriber subscriber) {
+        Set<LogReplicationSession> sessionForSinkSide = this.createOutgoingSessionsBySubscriber(subscriber);
+        for (LogReplicationSession session : sessionForSinkSide) {
```

Comment:
We don't need this part. 
Once we see that a subscriber has registered, we create outgoing sessions, and then we start the FSM. The FSM is going to send leadership query as part of its "VerifyingRemoteSinkLeaderState" state.

so instead of calling createAndSendOutgoingSession, you can directly call createSession(), but add a "subscriber" parameter to it. 
Also, the logic to create incoming sessions in createSession() needs to be removed.

Reply:
1. After some debugging, for the session that contains client information, I found out that "VerifyingRemoteSinkLeaderState" was indeed called, but the code path didn't touch "onEntry", then "verifyRemoteLeader" was not called. It seems that the leadership query will not be sent out in this case. I think we can put this method into createSession() though.
2. Removing the logic to create incoming sessions causes IT failures in LogReplicationRoutingQueueIT when createSession() is called from refresh(). Do you think it is a good idea to pass a boolean as the parameter of createSession(), and to see if createSession() is called from refresh(), create incoming sessions, if it is called from handleRegistrationTableEntries(), do not create incoming sessions?

Reply:
Need to figure out why the why "verifyRemoteLeader" was not called.